### DISCUSSION TITLE
Problem With Hot Reload

### DISCUSSION THREAD

**Discussion by jcantrel** [tailwindlabs/tailwindcss#6931] [Help]
I do not know if this is actually possible tp fix in 3.0 due to the way all classes are available out of the box, but felt it was worth establishing as an issue for further work.

I use Tailwind with Blazor .net 6, and one of the biggest advantages of .net 6 is hot reload of code so I do not have to recompile every time I make a change. However with how 3.0 is JIT at all times this causes huge issues with all projects, but new ones especially. Since the values are purged on compile, if I am working on building a new page and using Tailwind CSS I have not used already hot reload will not be able to show the new CSS. I have to stop my debugger and recompile.

In 2.0 I could simply disable JIT until I was ready to publish, so all classes would be available. Something like a debug mode that can be turned on and off, with all the most important classes safelisted (grid, form inputs, colors, etc.) by default might lessen the issue. It would not really solve it though.

**Comment by jshimkoski** [comment]
Not sure if this is related to the original post but:

I have a nuxt 2 codebase with 150+ components (unable to provide source, sorry).

When I run `npm run dev` while using Tailwind CSS 3.0.10 or 3.0.11, if I change any file, an infinite loop in terminal fires off, as if the hot reload for nuxt 2 is picking up a changed file (random file name it seems) provided by Tailwind. The only way to stop the loop is to perform a hard kill (CTRL+C) in terminal.

If I downgrade to 3.0.9, the infinite loop goes away.

I used the instructions provided on the Tailwind CSS 3 reference guide to install Tailwind CSS.

**Reply by jshimkoski** [reply]
I just fixed the infinite loop issue in my Nuxt 2 app.

In `nuxt.config.js` I had both of these buildModules listed:

```
buildModules: [
  '@nuxtjs/eslint-module',
  '@nuxt/typescript-build'
]
```

**This is a big no no**. If you look at the [Nuxt Typescript docs](https://typescript.nuxtjs.org/guide/lint/), it states that you need to remove `@nuxtjs/eslint-module` when using `@nuxt/typescript-build`.

Once removed, the infinite loop issue was fixed.

I have no idea why Tailwind 3.0.10 and up triggered the infinite loop, but here we are. 🤷 

On a side note, this drastically sped up Hot Reload in my app as well.

I hope this helps others that encounter a similar issue.

**Comment by jcantrel** [comment]
As an update, this also affects intellisense in Visual Studio Professional. Since it is looking at my CSS file, which is trimmed down automatically at run time, it cannot be used for any Tailwind CSS that has not already been used. Just like hot reload cannot use it. This is causing significant slow down in development compared to when JIT could be disabled.

**Comment by MatthewKing** [comment]
@jcantrel Did you ever come up with a suitable solution for this? I've just discovered Tailwind and am hitting the same issues with my Blazor apps and hot reload.

**Comment by jcantrel** [comment]
The best option is to set up npx tailwindcss -i .\Styles\app.css -o .\wwwroot\css\app.css --watch for your dev folder. However, while I have seen this work for others it does not work on either of my dev machines. It works until I start debug mode, then the watch freezes and I have to ctrl+c it. Instead, when I need to update my CSS I have to run it manually. npx tailwindcss -i .\Styles\app.css -o .\wwwroot\css\app.css

**Comment by RemiCELLARD** [comment]
Hi, do you have any update about your issue @jcantrel @MatthewKing ? 
I have the same issue.

As you mentioned, I have to run it manually with `npx tailwindcss -i wwwroot/css/app-tailwind.css -o wwwroot/css/app-tailwind.min.css --watch` before i launch my blazor app.


Thanks
